src/SearchTrips/Trip.tsx

```diff
@@ -0,0 +1,156 @@
+      color: Color.Blue,
+    },
+  });
+  return accessories;
+};
+
+type Props = {
+  origin: Feature;
+  destination: Feature;
+  strategy: "arrive-by" | "leave-at";
+};
+export default function Trip({ origin, destination, strategy }: Props) {
+  const [trips, setTrips] = useState<TripPattern[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [pageCursor, setPageCursor] = useState("");
+  const [isDetailVisible, setDetailVisible] = useState(false);
+  const [dateTime] = useState(new Date());
+
+  const getTrips = useCallback(() => {
+    setIsLoading(true);
+    fetchTrip({
+      originId: origin.properties.id,
+      destinationId: destination.properties.id,
+      arriveBy: strategy === "arrive-by",
+      dateTime,
+      pageCursor,
+    })
+      .then((data) => {
+        const newTrips = trips?.concat(data.trip.tripPatterns);
+        setTrips(newTrips);
+        setPageCursor(data.trip.nextPageCursor);
+      })
+      .finally(() => setIsLoading(false));
+  }, [pageCursor]);
+
+  useEffect(() => {
+    getTrips();
+  }, []);
+
+  return (
+    <List
+      isLoading={isLoading}
+      isShowingDetail={isDetailVisible}
+      searchBarPlaceholder={`From ${origin.properties.name} to ${destination.properties.name}...`}
+    >
+      {trips?.map((trip, idx) => (
+        <List.Item
+          detail={<List.Item.Detail metadata={<TripDetails trip={trip} />}></List.Item.Detail>}
+          actions={
+            <ActionPanel>
+              <Action
+                title="Toggle Details"
+                onAction={() => {
+                  setDetailVisible(!isDetailVisible);
+                }}
+              />
+              <Action
+                title="Load More"
+                onAction={() => {
+                  getTrips();
+                  /**
+                   * Perhaps we could remove this in favor of
+                   * `pagination` returned from using the built-in useFetch/usePromise
+                   * [more here](https://developers.raycast.com/utilities/react-hooks/usefetch#pagination)
+                   *
+                   */
+                }}
+              />
+            </ActionPanel>
+          }
+          title={`${formatAsClock(trip.expectedStartTime)} - ${formatAsClock(trip.expectedEndTime)}`}
+          // TODO: subtitle => "Little warning triangle if service disruptions are present"
+          accessories={buildAccessories(trip, isDetailVisible)}
+          key={idx}
+        />
+      ))}
+    </List>
+  );
+}
+
+const TripDetails = ({ trip }: { trip: TripPattern }) => {
+  const getTitleText = (leg: Leg, isArriveText?: boolean) => {
+    if (isArriveText) {
+      return `${formatAsClock(leg.expectedEndTime)} ${leg.toPlace.quay.name} ${leg.toPlace.quay.publicCode || ""}`;
+    }
+    if (leg.fromPlace.name === leg.toPlace.name) {
+      return `${formatAsClock(leg.expectedStartTime)} ${leg.fromPlace.quay.name}`;
+    }
```

Comment:
Not sure I understand why we're hiding `publicCode` in this case 🤔 Can these lines be removed?

Reply:
Only when it's a transfer between the same places. which is walking. but I guess when we have the quay, it makes sense to mention the publicCode. I've removed it 💪